src/vss_tools/vspec/vssexporters/vss2samm/helpers/vssHelper.py

```diff
@@ -0,0 +1,569 @@
+        #   Vehicle.Cabin.Seat.Occupant.Identifier.Issuer
+        # and
+        #   Vehicle.Driver.Identifier.Issuer
+        #
+        # In this case we need to take current vss_node parent ttl_name, instead of just its name,
+        # so this will make the current vss_node ttl_name unique enough,
+        # in order it can be loaded across different models.
+        #
+        # This is specially, when a user is using the split option
+        if vss_node.parent.name != vss_node.parent.ttl_name and \
+            top_vss_tree_unique_node_names[vss_node.name]["counter"] > 0 and \
+            sum(vss_node.parent.name in vss_path
+                for vss_path in top_vss_tree_unique_node_names[vss_node.name]["vss_paths"]) > 1:
+            parent_prefix = vss_node.parent.ttl_name
+
+        else:
+            parent_prefix = vss_node.parent.name
+
+        if ttl_name.startswith(parent_prefix):
+            return get_parent_prefix_for_ttl_name(vss_node.parent, ttl_name, use_vehicle_prefix=True)
+
+    return parent_prefix
+
+
+# Set ttl_name for provided vss_node.
+# BY DEFAULT: ttl_name will be set to the current vss_node.name.
+#   vss_node           - the VSSNode to be updated
+#   use_parent_prefix  - when provided, the vss_node.parent.name will be used as prefix to its vss_node.name,
+#                        if the vss_node has a parent
+#   overwrite_ttl_name - if specified and the provided vss_node, already has a ttl_name,
+#                        it will be overwritten based on preserve_ttl_name
+#   preserve_ttl_name  - if specified and the provided vss_node, already has a ttl_name,
+#                        then its ttl_name will be preserved and will be prefixed with its parent name
+def set_ttl_name(vss_node: VSSNode, use_parent_prefix: bool, overwrite_ttl_name=False, preserve_ttl_name=False):
+
+    if vss_node.ttl_name and not overwrite_ttl_name:
+        log.debug(
+            "Node ttl name: '%s' was already set.",
+            vss_node.ttl_name
+        )
+
+    else:
+        ttl_name = vss_node.ttl_name if vss_node.ttl_name and preserve_ttl_name else vss_node.name
+
+        if use_parent_prefix:
+            parent_prefix = get_parent_prefix_for_ttl_name(vss_node, ttl_name)
+            vss_node.ttl_name = parent_prefix + ttl_name
+
+        else:
+            vss_node.ttl_name = ttl_name
+
+        log.debug(
+            "%s node ttl name: '%s'.",
+            "updated" if overwrite_ttl_name else "added",
+            vss_node.ttl_name
+        )
+
+
+# Helper function to clone a provided VSSNode, instead of to copy it, using the python available copy functionality,
+# which will copy the whole node.
+#
+# This is required so we can have a "clean" unbound to properties of the provided VSSNode, copy of it,
+# which then can be further manipulated.
+#
+# For example: filter the children of cloned VSSNode and add them to its cloned (copied) version.
+# REASON FOR THIS is that:
```

Comment:
I don't think this is the case when e.g. using copy.deepcopy().  However, the performance of deepcopy is not great since it will essentially copy the complete tree including all parents and therefore all children etc. This is  one of the reasons I implemented an  own `copy`  method: https://github.com/COVESA/vss-tools/pull/382/files#diff-156f359c987ddf2af79c8b9bf50be3c1899d7a3e389865b0fe7b89812d62bc51R69.
Maybe we will merge the refactoring this week. That will simplify a lot of things but will cause  conflicts with this one unfortunately

Reply:
I was not aware of this.
Reason for this copy was explained in the comment above this: clone_vss_node function.
This was needed when this exporter was created as external tool which was using provided by vss-tools functionality so I am open for improvements if there is a way to reduce this effort for filtering of selected signals / branches etc.